runtime/str.go

```diff
@@ -753,6 +753,12 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 					return nil, raised
 				}
 				val = s.Value()
+			} else if matches[7] == "o" {
+				if o.isInstance(LongType) {
+					val = toLongUnsafe(o).Value().Text(8)
```

Comment:
@trotterdylan The behavior is weird.
```
class A(object):
  def __oct__(self):
    return '0123'

class B(object):
  def __oct__(self):
    return '0124'
  def __int__(self):
    return 8

# This line will cause a TypeError
# TypeError: %o format: a number is required, not A
# print '%o' % A()

# This line is correct
print '%o' % B()
```
Obviously this conversion doesn't use `__hex__` and `__oct__` now.